packages/suite/src/services/metadata/DropboxProvider.ts

```diff
@@ -40,41 +82,93 @@ class DropboxProvider implements AbstractMetadataProvider {
     }
 
     async getFileContent(file: string) {
-        const exists = await this.client.filesSearch({
-            path: '',
-            query: `${file}.mtdt`,
-        });
-        if (exists && exists.matches.length > 0) {
-            const file = await this.client.filesDownload({
-                path: exists.matches[0].metadata.path_lower!,
+        try {
+            const exists = await this.client.filesSearch({
+                path: '',
+                query: `${file}.mtdt`,
             });
-            // @ts-ignore: fileBlob not defined?
-            const buffer = await file.fileBlob.arrayBuffer();
-            return buffer;
+            if (exists?.matches?.length > 0) {
+                const file = await this.client.filesDownload({
+                    path: exists.matches[0].metadata.path_lower!,
+                });
+                // @ts-ignore: fileBlob not defined?
+                const buffer = (await file.fileBlob.arrayBuffer()) as Buffer;
+
+                return this.ok(buffer);
+            }
+            return this.ok(undefined);
+            // not found. this is not error. user just has not created the file yet
+        } catch (err) {
+            // example:
+            // {"error_description": "refresh token is malformed", "error": "invalid_grant"}
+            return this.handleProviderError(err);
         }
     }
 
     async setFileContent(file: string, content: Buffer) {
-        const blob = new Blob([content], { type: 'text/plain;charset=UTF-8' });
-        return this.client.filesUpload({
-            path: `/Apps/TREZOR/${file}.mtdt`,
-            contents: blob,
-            // @ts-ignore "overwrite" !== string?
-            mode: 'overwrite',
-        });
-    }
+        try {
+            const blob = new Blob([content], { type: 'text/plain;charset=UTF-8' });
+            await this.client.filesUpload({
+                path: `/Apps/TREZOR/${file}.mtdt`,
+                contents: blob,
+                // @ts-ignore
+                mode: 'overwrite',
+            });
 
-    isConnected() {
-        return this.connected;
+            return this.ok();
+        } catch (err) {
+            return this.handleProviderError(err);
+        }
     }
 
     async getCredentials() {
-        const account = await this.client.usersGetCurrentAccount();
-        return {
-            type: 'dropbox',
-            token: this.client.getAccessToken(),
-            user: account.name.given_name,
-        } as const;
+        const token = this.client.getRefreshToken();
+        if (!token) return this.error('AUTH_ERROR', 'token is missing');
```

Comment:
Is this error displayed to the user? If yes it should be localized.

Reply:
there are few more intl problems. I would like to fix them after the freeze so that we don't stop it from happening.